jib-core/src/main/java/com/google/cloud/tools/jib/builder/configuration/BuildConfiguration.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.builder.configuration;
+
+import com.google.cloud.tools.jib.builder.BuildLogger;
+import com.google.cloud.tools.jib.image.ImageReference;
+import com.google.cloud.tools.jib.registry.credentials.RegistryCredentials;
+import com.google.common.collect.ImmutableSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Immutable configuration options for the builder process. */
+public class BuildConfiguration {
```

Comment:
I was thinking this is more just a simple value/builder class?

Is there a reason we can't just use something like

```
public class BuildConfiguration {
   ImageReference imageReference;
   ... getImageReference()

   builder {
     build() {
        // validate all required params
        if (imageReference == null) -> then print out message?
    }

```

Reply:
Looks like I overcomplicated this haha.